src/borg/repository.py

```diff
@@ -1475,6 +1476,22 @@ def delete_segment(self, segment):
         except FileNotFoundError:
             pass
 
+    def clear_empty_dirs(self):
+        """Delete empty segment dirs, i.e those with no segment files.
+        """
+        data_dir = os.path.join(self.path, 'data')
+        segment_dirs = (f.path for f in os.scandir(data_dir) if f.is_dir() and f.name.isdigit())
```

Comment:
guess this should also call the code of #7606.

Reply:
Yes. Just give me a moment.  Rebasing. :)

Reply:
Done